turbopuffer/namespace.py

```diff
@@ -253,5 +314,113 @@ def recall(self, num=20, top_k=10) -> float:
         """
 
         response = self.backend.make_api_request('vectors', self.name, '_debug', 'recall', query={'num': num, 'top_k': top_k})
-        assert 'recall' in response, f'Invalid recall() response: {response}'
-        return float(response.get('recall'))
+        content = response.get('content', dict())
+        assert 'recall' in content, f'Invalid recall() response: {response}'
+        return float(content.get('recall'))
+
+
+class NamespaceIterator:
+    """
+    The VectorResult type represents a set of vectors that are the result of a query.
+
+    A VectorResult can be treated as either a lazy iterator or a list by the user.
+    Reading the length of the result will internally buffer the full result.
+    """
+
+    backend: Backend
+    namespaces: List[Namespace] = []
+    index: int = -1
+    offset: int = 0
+    next_cursor: Optional[Cursor] = None
+
+    def __init__(self, backend: Backend, initial_set: Union[List[Namespace], List[dict]] = [], next_cursor: Optional[Cursor] = None):
+        self.backend = backend
+        self.index = -1
+        self.offset = 0
+        self.next_cursor = next_cursor
+
+        if len(initial_set):
+            if isinstance(initial_set[0], Namespace):
+                self.namespaces = initial_set
+            else:
+                self.namespaces = NamespaceIterator.load_namespaces(backend.api_key, initial_set)
+
+    def load_namespaces(api_key: Optional[str], initial_set: List[dict]) -> List[Namespace]:
+        output = []
+        for input in initial_set:
+            ns = tpuf.Namespace(input['id'], api_key=api_key)
+            ns.metadata = {
+                'exists': True,
+                'dimensions': input['dimensions'],
+                'approx_count': input['approx_count'],
+                'created_at': input['created_at'],
```

Comment:
I think we need to parse this into a proper Python datetime

Reply:
I'm actually considering removing this for now, since until it's something we can get from the HEAD endpoint, we can't really expose a function for it.
The only way to populate this AFAIK is to use the list namespaces endpoint.

Reply:
@xthexder you could add it to HEAD if you want